### ISSUE TITLE
modeladmin select multiple for mass-action

### ISSUE THREAD

**Issue by nutztherookie** [torchbox/wagtail#2719]
hey torchies,

we are trying your new ModelAdmin-feature and realized there is no "I'm selecting a bunch of objects and I'm gonna <delete/export/dosomething> them all". Django Admin:
![image](https://cloud.githubusercontent.com/assets/320080/16110319/76ad85b0-33ad-11e6-82b7-b11e0b62721a.png)

So we were wondering: Are you planning on implementing this? If so, do you know when that might be the case? :) 

Thanks a bunch

PS: Great job on the ModelAdmin and Wagtail in general

**Comment by ababic** [comment]
Hi @nutztherookie! Thanks a lot for the feedback :)

I'd certainly love to see something that replicated Django ModelAdmin's 'actions' in the future, but there are no set plans to tackle that right away. I guess we have some thinking/decision-making to do, in regards to how this app evolves and becomes more integrated into the wagtail core (and what things in core need to also change, in order for that to happen)

When `modeladmin` was still in it's infancy (as [wagtailmodeladmin](https://github.com/rkhleics/wagtailmodeladmin)), I made an attempt to add an 'actions-like' feature, but at the time, I stumbled when it came to handling permissions, and ensuring only users with the right actions could perform them. I think it was particularly complicated because of the way pages have their own tree-based permission system, and I decided it was a little too complicated to tackle on my own. Hopefully now, with a keen group of wagtail contributors behind it, that's something that might be overcome.

**Comment by nutztherookie** [comment]
Hey @ababic, thank you for your elaborate answer =)

I know little of wagtails permissions-setup; but also I don't really need it for my project.
I'm gonna start hacking 'actions' into wagtail and I'll see where it leads me. If it turns out to be somewhat mature I'll try and upstream it :)

**Comment by SalahAdDin** [comment]
But, if we'll have this features, `django-admin` will be 'deprecated'. So, for now we can used `django-admin`normally.

**Comment by ababic** [comment]
@salahAdDin I'm afraid I'm struggling to understand what you mean there. Are you suggesting that this feature should or shouldn't be implemented? What do you mean when you say django admin will be 'deprecated'?  I'm pretty sure django admin isn't going anywhere. It's a fundamental part of django.

**Comment by SalahAdDin** [comment]
Ok, i say, for our use, with `wagtail-admin`and `model-admin`, for us, will be deprecated because, for what have `django-admin` if `wagtail-admin` do all?

**Comment by ababic** [comment]
Sorry @SalahAdDin, I don't know what you're asking. Nobody is planning to make any changes that would prevent you running django-admin and wagtailadmin for the same project if you wanted to. Having a feature like this would obviously give developers less need to do that though, which can only be a good thing.

**Comment by tomdyson** [comment]
Hi @nutztherookie did you get anywhere with 'actions'? A few other people have asked about this recently.

**Comment by nutztherookie** [comment]
hey @tomdyson, no sorry - i got derailed and ended up working on other things of the project.

i'd still like to work on it, but don't know when that might be.

**Comment by MechanisM** [comment]
Any moves forward on this? I need this one too.

**Comment by grwhumphries** [comment]
Just pinging this forward - has there been any implementation of this for Wagtail yet?  It can be done in the standard django-admin, so not the end of the world, but it would be really nice to have the option in Wagtail as well.

**Comment by devRawnie** [comment]
Re-visiting this in 2021. Would love such a feature, with the option to define actions in a subclass of ModelAdmin class, and apply those to multiple instances.

**Comment by SidSidSid16** [comment]
Having the ability to perform mass actions on multiple instances would make things so much easier. Would love an update on this.

**Comment by lb-** [comment]
Now that bulk actions exist as of 2.15 I will rename this to be add bulk actions to ModelAdmin

**Comment by alittle-ninja** [comment]
any update here?

**Comment by lb-** [comment]
@alittle-ninja there are no plans in place for this on the core roadmap at this time but anyone is welcome to prepare are PR for this feature.

**Comment by ababic** [comment]
On the topic of new features for modeladmin, I think we need to be careful how we tread, as there are rough plans to sunset that module completely (I'd hate for someone to go to all the effort of contributing and polishing, only for the work never to be merged). 

The plan isn't entirely clear, but from the conversations I've had with @laymonage, it looks something like this:

1. Continue to bring snippets functionality in-line with ModelAdmin in terms of features
2. Ensure things are designed in such  a way that most of the views and features from snippets are available on generic sets of views (of which, 'snippets' will be one opinionated 'flavour'). 
3. Figure out what the registration mechanisms look like for non-snippet models.
4. After a standard deprecation period (long enough for devs to translate old modeladmin code over to the new), either remove the modeladmin app completely, or move it out to it's own third-party app to die a bit more slowly 😜

@laymonage please do let me know if this summary is inaccurate. It's been a while since we discussed this!

**Comment by laymonage** [comment]
The summary posted by @ababic above is accurate.

In regards to bulk actions, I think it could use some refactoring internally before we start expanding it, as the approach is not ideal in some places, e.g. how the strings are set up for the client-side interface: https://github.com/wagtail/wagtail/blob/a6a91e1a014912c6f7f65cc367c099e9338c8906/wagtail/admin/localization.py#L58-L95

Anyway, as per wagtail/rfcs#85, I'm transferring this issue over to the wagtail-modeladmin repo.
